service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManagerOptions.java

```diff
@@ -9,9 +9,18 @@
 @Builder
 public class RegistryManagerOptions
 {
+    protected static final Integer DEFAULT_HTTP_READ_TIMEOUT_MS = 24000; // 24 seconds
+    protected static final Integer DEFAULT_HTTP_CONNECT_TIMEOUT_MS = 0; // infinite
```

Comment:
why infinite connect timeout?

Reply:
This was the default behavior since the SDK never set an http connect timeout before.

Reply:
By default, http connections in java have no read or connect timeouts. However, our SDK did manually set the read timeout to 24 seconds. This PR doesn't change the default behavior of the SDK

Reply:
Sure, but should the SDK set a timeout value for http connect? If the http connection cannot be established (network issue/ proxy hasn't been set up etc), then when would this actually time out and throw an exception? Would we reply on the TCP layer underneath throwing an exception (which can take minutes)?

Reply:
One could argue it wouldn't be a breaking change to start adding connect timeouts by default. The same exception would be thrown for a connect timeout as a read timeout. The only difference is the exception message. I'll add the same default of 24 seconds for the connect timeout